controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -56,13 +76,57 @@ func (r *K8ssandraClusterReconciler) reconcileMedusa(
 			}
 			secret.AddInjectionAnnotationMedusaContainers(&dcConfig.Meta.Pods, cassandraUserSecretName)
 		}
+
+		// Reconcile the Medusa standalone deployment
+		kcKey := utils.GetKey(kc)
+		desiredMedusaStandalone.SetLabels(labels.CleanedUpByLabels(kcKey))
```

Comment:
RE CleanedUpByLabels, would the medusa deployment not always be in the same namespace as the K8ssandraCluster and therefore be cleaned up using standard k8s GC and OwnerReferences? 

Regardless, I understand that PartOf labels ought to be applied to identify the cluster, cluster-home-namespace that the deployment belongs to. However, I am confused as to why we're using labels based cleanup in this instance.

Additionally, I also wonder whether this feedback may also apply to the services and deployments cleanup code too. (Sorry, this only just occured to me :/ )